### DISCUSSION TITLE
Guidance how to integrate with others WebAssembly library

### DISCUSSION THREAD

**Discussion by vdiallonort** [AvaloniaUI/Avalonia#16515] [Q&A]
Hello,
I am starting my journey with Avalonia and checking if it's the right technology for my app. I am looking to build a ChatBot app with Avalonia using LLM.The whole app needs to be executed in the browser (no backend for security reasons). There is a web Assembly for LLM named WebLLM [](https://github.com/mlc-ai/web-llm). Is it possible to integrate this into a Wasm Avalonia app? Or will it be a bad fit?

**Comment by maxkatz6** [comment]
You can reference JavaScript or WASM code in .NET for Browser:
https://learn.microsoft.com/en-us/aspnet/core/blazor/webassembly-native-dependencies?view=aspnetcore-8.0&preserve-view=true
https://learn.microsoft.com/en-us/aspnet/core/blazor/javascript-interoperability/import-export-interop?view=aspnetcore-8.0
